src/pie_models/models/components/pooler.py

```diff
@@ -0,0 +1,157 @@
+import logging
+from typing import Any, Callable, Dict, Tuple
+
+import torch
+from torch import Tensor, cat, nn
+
+# possible pooler types
+CLS_TOKEN = "cls_token"  # CLS token
+START_TOKENS = "start_tokens"  # MTB start tokens concat
+MENTION_POOLING = "mention_pooling"  # mention token pooling and concat
+
+
+logger = logging.getLogger(__name__)
+
+
+def pool_cls(hidden_state: Tensor, **kwargs) -> Tensor:
+    return hidden_state[:, 0, :]
+
+
+class AtIndexPooler(nn.Module):
+    """Pooler that takes the hidden state at a given index.
+
+    Args:
+        input_dim: The input dimension of the hidden state.
+        num_indices: The number of indices to pool.
+        offset: The offset to add to the indices.
```

Comment:
what is the var offset for? (i.e. what are the semantics or why is it necessary?)

Reply:
addressed in [46224f8](https://github.com/ArneBinder/pie-models/pull/3/commits/46224f855b74b6b412b53e0721a40377412d5f7a)